deployment/ccip/changeset/cs_chain_contracts.go

```diff
@@ -1685,3 +1681,128 @@ func DefaultFeeQuoterDestChainConfig(configEnabled bool, destChainSelector ...ui
+			return err
+		}
+	}
+	for chainSel, updates := range cfg.UpdatesByChain {
+		if err := ValidateChain(e, state, chainSel, cfg.MCMSConfig); err != nil {
+			return err
+		}
+		chainState := state.Chains[chainSel]
+		if chainState.FeeQuoter == nil {
+			return fmt.Errorf("missing fee quoter for chain %d", chainSel)
+		}
+		tokenAddresses, err := chainState.TokenAddressBySymbol()
+		if err != nil {
+			return fmt.Errorf("error getting token addresses for chain %d: %w", chainSel, err)
+		}
+		for _, token := range updates.TokensToRemove {
+			if _, ok := tokenAddresses[token]; !ok {
+				return fmt.Errorf("token %s not found in state for chain %d", token, chainSel)
+			}
+		}
+		for _, token := range updates.TokensToAdd {
+			if _, ok := tokenAddresses[token]; !ok {
+				return fmt.Errorf("token %s not found for in state chain %d", token, chainSel)
+			}
+		}
+		if err := commoncs.ValidateOwnership(
+			e.GetContext(),
+			cfg.MCMSConfig != nil,
+			e.Chains[chainSel].DeployerKey.From,
+			state.Chains[chainSel].Timelock.Address(),
+			state.Chains[chainSel].FeeQuoter,
+		); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func ApplyFeeTokensUpdatesFeeQuoterChangeset(e deployment.Environment, cfg ApplyFeeTokensUpdatesConfig) (deployment.ChangesetOutput, error) {
+	if err := cfg.Validate(e); err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	state, err := LoadOnchainState(e)
+	if err != nil {
+		return deployment.ChangesetOutput{}, err
+	}
+	var batches []mcmstypes.BatchOperation
+	timelocks := make(map[uint64]string)
+	proposers := make(map[uint64]string)
+	inspectorPerChain := map[uint64]sdk.Inspector{}
+	for chainSel, updates := range cfg.UpdatesByChain {
+		chain := e.Chains[chainSel]
+		txOpts := e.Chains[chainSel].DeployerKey
+		if cfg.MCMSConfig != nil {
+			txOpts = deployment.SimTransactOpts()
+		}
+		fq := state.Chains[chainSel].FeeQuoter
+		tokenAddresses, err := state.Chains[chainSel].TokenAddressBySymbol()
+		if err != nil {
+			return deployment.ChangesetOutput{}, fmt.Errorf("error getting token addresses for chain %d: %w", chainSel, err)
+		}
+		var tokensToRemove, tokensToAdd []common.Address
+		for _, token := range updates.TokensToRemove {
+			tokensToRemove = append(tokensToRemove, tokenAddresses[token])
+		}
+		for _, token := range updates.TokensToAdd {
+			tokensToAdd = append(tokensToAdd, tokenAddresses[token])
+		}
+		tx, err := fq.ApplyFeeTokensUpdates(txOpts, tokensToRemove, tokensToAdd)
+		if cfg.MCMSConfig == nil {
+			if _, err := deployment.ConfirmIfNoErrorWithABI(e.Chains[chainSel], tx, fee_quoter.FeeQuoterABI, err); err != nil {
+				return deployment.ChangesetOutput{}, fmt.Errorf("error applying token updates for chain %d: %w", chainSel, err)
+			}
+		} else {
+			if err != nil {
+				return deployment.ChangesetOutput{}, err
+			}
+			op, err := proposalutils.BatchOperationForChain(
+				chainSel, fq.Address().String(), tx.Data(), big.NewInt(0), cfg.MCMSConfig.MCMSType.String(), nil)
```

Comment:
I think we want the changeset contract type here, in your case FeeQuoter
```suggestion
				chainSel, fq.Address().String(), tx.Data(), big.NewInt(0), string(FeeQuoter), nil)
```